src/v/storage/log_manager.cc

```diff
@@ -122,6 +122,7 @@ log_manager::housekeeping_scan(model::timestamp collection_threshold) {
         co_await current_log.handle.compact(compaction_config(
           collection_threshold,
           _config.retention_bytes(),
+          current_log.handle.stm_manager()->max_compactible_offset(),
```

Comment:
it kind of seems like rm_stm should update storage layer (as opposed to storage querying rm_stm). rm_stm is in a position to know when an offset changes, and having information flow downward there is less complex coupling. wdyt?

Reply:
@mmaslankaprv suggested using polling, it seems like a more straightforward solution, also in the next part of the compaction /transaction integration the compaction logic will need to query if a given record is a part of the aborted or committed transaction. using an object to query instead of passing a list of aborted records gives more space for memory optimization

Reply:
I wonder if we are starting to approach a situation where we need to think about moving part of the transaction management into the storage layer.

Reply:
> @mmaslankaprv suggested using polling, it seems like a more straightforward solution

I'm not sure I agree it is more straight forward when we consider the lifetime of all the new objects involved. But I do agree for querying and memory optimization of aborted records it doesn't make sense to push that down. However, I do think that suggests some reorganization would be beneficial.

Reply:
I agree that the inverted dependencies are annoying and it's a good idea to lift compaction logic above raft and let stm directly control it but it's out of scope of the compaction / transactions integration and we can't condition transactions release on the fundamental refactoring of the compaction logic.

Right now we compact adjacent segments but when the transactions are involved in order to decide if a record b overwrites a record a (even when the records are from the same segment) we may need to look ahead for a unbounded amount of segments for search for corresponding abort or commit txn markers. So we either need to change compaction from working with adjacent segments to work with unbounded segments or to rely on an external oracle who happens to know future to guide the compaction logic. The oracle requires less changes.

But there is a silver lining - we are not increasing the tech debt balance. The storage layer already depends on stm_manager which contains a reference to rm_stm so things are not getting worse.

Reply:
I think i agree with Noah here, it would be cleaner to reverse the dependency, but i also got Denis'es point about already existing `stm_manager`

Reply:
Yep. I was thinking similar things as I wrote this PR; would like to chat about layering here.

Reply:
> But there is a silver lining - we are not increasing the tech debt balance. The storage layer already depends on stm_manager which contains a reference to rm_stm so things are not getting worse.

thanks for mentioning this observation